Lib/traceback.py

```diff
@@ -1095,6 +1095,30 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
                         self._str += f" Or did you forget to import '{wrong_name}'?"
                     else:
                         self._str += f". Did you forget to import '{wrong_name}'?"
+        elif exc_value and isinstance(exc_value, UnpackError):
+            if exc_value.msg is not None:
+                self._str = exc_value.msg
+            else:
+                lhs_length = exc_value.expected_count
+                self._str = f"too many values to unpack (expected {lhs_length})"
+                try:
+                    rhs_length = len(exc_value.iterable)
```

Comment:
By keeping a reference on the iterable in the exception, could we have possible reference cycles? as far as I can tell, you're only interested in the iterable's length, if any, so why not do it at the C level?

Reply:
Forget the precomputation point but please confirm that we wouldn't have issues by storing a generator in an exception for whatever reason.